static-mock/static_mock_service.go

```diff
@@ -137,6 +139,52 @@ func loadStaticMockRequestsAndResponses(wiretapService *daemon.WiretapService, l
 	return staticMockDefinitions
 }
 
+// StartWatcher Function to start a watcher on mock-definitions folder
+func (sms *StaticMockService) StartWatcher() {
+	pathToWatch := sms.wiretapService.StaticMockDir + MockDefinitionsPath
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		sms.logger.Error("Error when creating fsnotify.NewWatcher '%s'", err.Error(), err)
+	}
+
+	err = watcher.Add(pathToWatch)
```

Comment:
I would add a nill / empty string check here and return early if there is nothing to watch.

Reply:
In fact, the early return should be on line 145